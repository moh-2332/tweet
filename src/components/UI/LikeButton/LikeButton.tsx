import React from "react";

const LikeButton: React.FC<{
  isFavorite: boolean;
  onToggle: (toggle: boolean) => void;
}> = ({ isFavorite, onToggle }) => {
  const toggleHandler = () => {
    onToggle(!isFavorite);
  };

  return (
    <span className="material-icons" onClick={toggleHandler}>
      {isFavorite ? "favorite" : "favorite_border"}
    </span>
  );
};

export default LikeButton;
