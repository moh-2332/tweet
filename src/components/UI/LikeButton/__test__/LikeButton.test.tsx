import React from "react";

import { screen, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

import LikeButton from "../LikeButton";

const handler = jest.fn();

describe("LikeButton tests", () => {
  it("Should match the snapshot", () => {
    const likeButton = renderer.create(
      <LikeButton isFavorite={false} onToggle={handler} />
    );
    expect(likeButton).toMatchSnapshot();
  });

  it("Should renders the empty favorite icon", () => {
    render(<LikeButton isFavorite={false} onToggle={handler} />);

    const isNotLikedIcon = screen.getByText(/favorite_border/i);
    expect(isNotLikedIcon).toBeInTheDocument();
  });

  it("Should renders the favorite icon", () => {
    render(<LikeButton isFavorite={true} onToggle={handler} />);

    const isLikedIcon = screen.getByText(/favorite/i);
    expect(isLikedIcon).toBeInTheDocument();
  });

  it("Should toggle the favorite icon", () => {
    render(<LikeButton isFavorite={false} onToggle={handler} />);

    const isNotLikedIcon = screen.getByText(/favorite_border/i);
    expect(isNotLikedIcon).toBeInTheDocument();

    userEvent.click(isNotLikedIcon);

    const isLikedIcon = screen.getByText(/favorite/i);
    expect(isLikedIcon).toBeInTheDocument();
  });
});
