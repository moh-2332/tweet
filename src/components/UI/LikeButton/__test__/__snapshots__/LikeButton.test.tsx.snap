// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeButton tests Should match the snapshot 1`] = `
<span
  className="material-icons"
  onClick={[Function]}
>
  favorite_border
</span>
`;
